wasm/src/types/boolean.rs

```diff
@@ -0,0 +1,187 @@
+// Copyright (C) 2019-2025 Provable Inc.
+// This file is part of the Provable SDK library.
+
+// The Provable SDK library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The Provable SDK library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with the Provable SDK library. If not, see <https://www.gnu.org/licenses/>.
+
+use crate::{
+    Plaintext,
+    from_js_typed_array,
+    to_bits_array_le,
+    types::native::{BooleanNative, LiteralNative, PlaintextNative},
+};
+use snarkvm_console::prelude::{Double, FromBits, FromBytes, One, Pow, ToBits, ToBytes, Zero};
+use snarkvm_wasm::utilities::Uniform;
+
+use js_sys::{Array, Uint8Array};
+use once_cell::sync::OnceCell;
+use std::{ops::Deref, str::FromStr};
+use wasm_bindgen::prelude::*;
+
+/// Boolean element.
+#[wasm_bindgen]
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct Boolean(BooleanNative);
+
+
+#[wasm_bindgen]
+impl Boolean {
```

Comment:
This function should have a constructor that takes a boolean and uses the `new` method on the native type.